docs/contributing/contributing.md

```diff
@@ -72,11 +65,52 @@ To create binaries:
 
 GHC 8.6.5 is not supported here because `nixpkgs-unstable` no longer maintains the corresponding packages set.
 
-## Introduction tutorial
+## Testing
 
-See the [tutorial](./plugin-tutorial.md) on writing a plugin in HLS.
+The tests make use of the [Tasty](https://github.com/feuerbach/tasty) test framework.
+
+There are two test suites in the main haskell-language-server package, functional tests, and wrapper tests.
+Other project packages, like the core library or plugins, can have its own test suite.
+
+### Testing with Cabal
+
+Running all the tests
+
+```bash
+$ cabal test
+```
+
+Running just the functional tests
+
+```bash
+$ cabal test func-test
+```
+
+Running just the wrapper tests
+
+```bash
+$ cabal test wrapper-test
+```
+
+Running a subset of tests
+
+Tasty supports providing
+[Patterns](https://github.com/feuerbach/tasty#patterns) as command
+line arguments, to select the specific tests to run.
+
+```bash
+$ cabal test func-test --test-option "-p hlint"
+```
 
-## Test your hacked HLS in your editor
+The above recompiles everything every time you use a different test option though.
+
+An alternative is
```

Comment:
Can you please give an explanation as to why the alternative is preferrable?
E.g. (not sure if that's true)

An alternative, which only recompiles when tests (or dependencies) change:

Reply:
```suggestion
An alternative, which only recompiles when tests (or dependencies) change:
```
That would be the proposed change?

Reply:
Yes, if that's true :)

Reply:
just checked it :slightly_smiling_face: